internal/cmd/auth.go

```diff
@@ -141,11 +145,58 @@ func authenticationGRPC(
 
 	// only enable enforcement middleware if authentication required
 	if authCfg.Required {
-		interceptors = append(interceptors, authmiddlewaregrpc.UnaryInterceptor(
+		if authCfg.Methods.JWT.Enabled {
+			authJWT := authCfg.Methods.JWT
+
+			var ks jwt.KeySet
+
+			if authJWT.Method.JWKSURL != "" {
+				ks, err = jwt.NewJSONWebKeySet(ctx, authJWT.Method.JWKSURL, "")
+				if err != nil {
+					return nil, nil, nil, fmt.Errorf("failed to create JSON web key set: %w", err)
+				}
+			} else if authJWT.Method.PublicKeyFile != "" {
+				keyPEMBlock, err := os.ReadFile(authJWT.Method.PublicKeyFile)
+				if err != nil {
+					return nil, nil, nil, fmt.Errorf("failed to read key file: %w", err)
+				}
+
+				publicKey, err := jwt.ParsePublicKeyPEM(keyPEMBlock)
+				if err != nil {
+					return nil, nil, nil, fmt.Errorf("failed to parse public key PEM block: %w", err)
+				}
+
+				ks, err = jwt.NewStaticKeySet([]crypto.PublicKey{publicKey})
+				if err != nil {
+					return nil, nil, nil, fmt.Errorf("failed to create static key set: %w", err)
+				}
+			}
+
+			validator, err := jwt.NewValidator(ks)
+			if err != nil {
+				return nil, nil, nil, fmt.Errorf("failed to create JWT validator: %w", err)
+			}
+
+			exp := jwt.Expected{
+				SigningAlgorithms: []jwt.Alg{jwt.RS256, jwt.RS512, jwt.ES256, jwt.ES512},
```

Comment:
Is this an exhaustive list?

Reply:
Its not, but it is intentionally restricted to a set of common assymmetric algorithms.
I think we could get a long way with just this set.
Potentially someone will want a symmetric option at some point (e.g. HS256/HS512), but probably can hold off on that until requested. Symmetric will require a space in configuration for a private key.

Reply:
Interestingly, the `cap/jwt` only supports asymmetric. I wonder if that is just all they're interested in using as Hashi.

Reply:
@GeorgeMac Thank you for the explanation, it makes sense!